### PR TITLE
fix: check if some shortcuts are triggerable

### PR DESCRIPTION
**Pull request by decodism**
Check if some of the shortcuts are triggerable, add a "Toggle fullscreen window" shortcut, and add a "New window" shortcut and thumbnail button.

The colors of the "New window" thumbnail button are rather placeholders.

### CODE REVIEW

**Comment by decodism** [comment]
"New window" thumbnail button:
new_window_thumbnail_button

**Comment by lwouis** [comment]
This PR does multiple things. Could you please explain what the "check if some shortcuts are triggerable" part intends to do? I checked and it adds checks that already exists:

* https://github.com/lwouis/alt-tab-macos/blob/master/src/logic/Window.swift#L105
* https://github.com/lwouis/alt-tab-macos/blob/master/src/logic/Window.swift#L118
* https://github.com/lwouis/alt-tab-macos/blob/69bfb4180c98a7cafc5f596705d7e53e1007f013/src/ui/App.swift#L140
* https://github.com/lwouis/alt-tab-macos/blob/69bfb4180c98a7cafc5f596705d7e53e1007f013/src/ui/App.swift#L136
* https://github.com/lwouis/alt-tab-macos/blob/69bfb4180c98a7cafc5f596705d7e53e1007f013/src/logic/Application.swift#L129

I'm confused why you're adding another layer of checks here. What were you trying to address?

**Comment by lwouis** [comment]
Regarding the other 3 commits, I feel sad that you didn't discuss them in some tickets before investing time in implementing them. I'm afraid these changes introduce use-cases that are too niche. They add complexity. For instance the new button on hover makes it too intimidating now for new users. I already thought a lot before adding the purple "quit app" button, but observed that many people wanted that fonctionnality.

In other words, I think in the balance of adding complexity vs supporting popular use-cases, I'm affraid these are tipping the scale the wrong way.

**Comment by lwouis** [comment]
> Adding these checks in the event handling allows an alert sound to be played if they don't pass.

This is quite interesting. It seems like a better way to handle the inputs than what we do today.

Then I think we should migrate the current checks to the more upstream location. That means removing them where they are today (I gave examples above). We need to review all the code paths and make you to capture all the various checks if we want to bring them upstream without missing any. These things can be quite subtle.

**Comment by decodism** [comment]
Scope change: simply play the alert sound when a shortcut is unavailable for a thumbnail.

**Comment by lwouis** [comment]
> @lwouis I still find it odd not to have a fullscreen shortcut.

I agree, it's weird, and it would probably be useful to some people. I lament the never ending list of preferences though :/

File: src/logic/Application.swift [hunk 1]
```diff
@@ -125,9 +125,12 @@ class Application: NSObject {
         }
     }
 
-    func quit() {
+    func quit(_ alertSound: Bool = false) {
         // only let power users quit Finder if they opt-in
-        if runningApplication.bundleIdentifier == "com.apple.finder" && !Preferences.finderShowsQuitMenuItem { return }
+        if runningApplication.bundleIdentifier == "com.apple.finder" && !Preferences.finderShowsQuitMenuItem {
+            if alertSound { NSSound.beep() }
```
[the remark this answers is not in the document]
Comment (decodism) [reply src/logic/Application.swift:131]: I'm not sure it's a good idea to move these checks upstream, and this solution seems more practical than duplicating them.

[the remark this answers is not in the document]
Comment (lwouis) [reply src/logic/Application.swift:131]: I tested, and there is another issue upstream: some shortcuts don't trigger the sound naturally. Modifiers-only shortcuts for instance don't trigger it. It's a similar situation with the key-repeat, where we create a `NSTimer` instead on relying on the natural key repeat because some modifiers don't repeat naturally, so we create it artificially.

So in the end I think you approach here is the best 👍

**Comment by lwouis** [comment]
I took this commit, and revisted the code a bit to reuse the logic between UI components and keyboard shortcuts. I'm closing this PR, and the commit will remain from you, and will be present in the next release 👍
